src/EnergyPlus/EvaporativeFluidCoolers.cc

```diff
@@ -108,7 +108,7 @@ namespace EvaporativeFluidCoolers {
 
         // Now look for this particular object
         for (auto &thisEFC : state.dataEvapFluidCoolers->SimpleEvapFluidCooler) {
-            if ((thisEFC.TypeOf_Num == objectType) && (thisEFC.Name == objectName)) {
+            if ((thisEFC.TypeOf_enum == static_cast<DataPlant::PlantEquipmentType>(objectType)) && (thisEFC.Name == objectName)) {
```

Comment:
Why TypeOf_enum and not just Type? Also, why is this static_cast necessary? Why is `thisEFC.Type` not already of type `DataPlant::PlantEquipmentType`?